global_helpers/panther_oss_helpers.py

```diff
@@ -328,6 +331,40 @@ def evaluate_threshold(key: str, threshold: int = 10, expiry_seconds: int = 3600
     return False
 
 
+def geoinfo_from_ip(ip: str) -> dict:  # pylint: disable=invalid-name
+    """Looks up the geolocation of an IP address using ipinfo.io
+
+    Example ipinfo output:
+    {
+        "ip": "108.28.166.251",
+        "city": "Arlington",
+        "region": "Virginia",
+        "country": "US",
+        "loc": "38.8810,-77.1043",
+        "postal": "22226",
+        "timezone": "America/New_York",
+        "readme": "https://ipinfo.io/missingauth"
+    }
+    """
+    valid_ip = ip_address(ip)
```

Comment:
what happens if an invalid ip is passed in here? exception? or None? or...? just wondering if we need to catch anything before passing this into requests

Reply:
@nhakmiller says let the rules engine handle errors generally. At least I thought that's what he told me.

Reply:
Yeah looks like this will raise a `ValueError` if `ip` is not a valid IP, which will get reported through the normal channels.

If this didn't raise an error, I suspect the requests call would return a  non-200 value which would get reported down below. So either way should be covered 👍